django/db/models/fields/related_descriptors.py

```diff
@@ -195,6 +195,13 @@ def get_prefetch_querysets(self, instances, querysets=None):
         else:
             query = {"%s__in" % self.field.related_query_name(): instances}
         queryset = queryset.filter(**query)
+        # Since there can be only one object prefetched for each instance
+        # order by is insignificant and slows down the query.
+        # queryset = queryset.order_by()
+        # queryset was cloned by filter() just above, no need to re-clone it.
+        queryset.query.clear_ordering()
+        # No need to handle the case where the query is_sliced (LIMIT)
+        # because no filter can be added to a sliced query.
```

Comment:
Such a detailed comment is unnecessary. I'd prefer Simon's suggestion:
```suggestion
        # There can be only one object prefetched for each instance so
        # clear ordering if the query allows it without side effects.
        queryset.query.clear_ordering()
```

Reply:
Done